--- gaeilge.aff	2018-07-20 10:18:45.414976055 -0500
+++ gaeilgemor.aff	2018-07-20 12:34:54.438003257 -0500
@@ -6,7 +6,7 @@
 nroffchars	().\\*
 texchars	()\[]{}<\>\\$*.%
 allaffixes on                         
-compoundwords off
+compoundwords on
 
 # headers:char-sets
 defstringtype "nroff" "nroff" ".mm" ".ms" ".me" ".man" ".NeXT"
@@ -73,6 +73,7 @@
 # h
 flag H:
  	[AEIOUÁÉÍÓÚ]	>	H
+ 	[AEIOUÁÉÍÓÚ]	>	"H-"
 	
 #  m'
 flag M:
@@ -332,6 +333,8 @@
 	[OÁÓÚ] I M		>	-M,GÍ        # ordaitheach
 	[OÁÓÚ] I M		>	-M,DÍS
 	[OÁÓÚ] I M		>	IS
+	[OÁÓÚ] I M		>	-IM,FAD      # An B.N.
+	[OÁÓÚ] I M		>	-IM,FAIR     # An B.N.
 
 						     # buaim, iaim
 	[IU] A I M		>	-IM,NN       # láithreach CB lch.98
@@ -344,6 +347,8 @@
 	[IU] A I M		>	-M,GÍ        # ordaitheach
 	[IU] A I M		>	IS
 	[IU] A I M		>	-M,DÍS
+	[IU] A I M		>	-IM,FAD       # An B.N.
+	[IU] A I M		>	-IM,FAIR      # An B.N.
 
 	E A I M			>	-AIM,ITEAR   # glaeaim->glaeitear
 
@@ -357,6 +362,8 @@
 	[^AOÁÉÓÚ] I M		>	-M,GÍ        # ordaitheach
 	[^AOÁÉÓÚ] I M		>	IS
 	[^AOÁÉÓÚ] I M		>	-M,DÍS
+	[^AOÁÉÓÚ] I M		>	-IM,FEAD     # An B.N.
+	[^AOÁÉÓÚ] I M		>	-IM,FIR      # An B.N.
 
 	[^HAOÁÉÓÚT] I M		>	-IM,TEAR     # láithreach saor
 
@@ -383,6 +390,8 @@
 	É I M		>	-M,GÍ                # ordaitheach
 	É I M		>	IS
 	É I M		>	-M,DÍS
+	É I M		>	-M,FEAD              # An B.N.
+	É I M		>	-M,FIR               # An B.N.
 
 						     # guím
 	Í M		>	-M,ONN               # láithreach CB lch.99
@@ -394,6 +403,8 @@
 	Í M		>	-M,GÍ                # ordaitheach 
 	Í M		>	IS
 	Í M		>	-M,DÍS
+	Í M		>	-M,FEAD              # An B.N.
+	Í M		>	-M,FIR               # An B.N.
 
 	[^AOU] Í M		>	-ÍM,ITEAR
   						     # cnaím, maím
@@ -425,6 +436,7 @@
 	[^IUF] A I N N		>	-AINN,FAIDÍS
 	[^IUF] A I N N		> 	-AINN,FAÍ
 	[^IU] A I N N		>	-INN,DAR     # An B.N.
+	[^IU] A I N N		>	-INN,S       # An B.N.
 
 	[^HEUTI] A I N N		>	-AINN,TÁ     # gnáthchaite
 	[^HEUTI] A I N N		>	-AINN,TAÍ     
@@ -479,6 +491,7 @@
 	[IU] A I N N		>	-INN,FAIDÍS
 	[IU] A I N N		> 	-INN,FAÍ
 	[IU] A I N N		>	-INN,DAR     # An B.N.
+	[IU] A I N N		>	-INN,S       # An B.N.
 
 	E A I N N		>	-AINN,ITEÁ   # ghlaeainn->ghlaeiteá
 	E A I N N		>	-AINN,ITÍ
@@ -495,6 +508,7 @@
 	[^AOÁÉÓÚ] I N N		>	-INN,FIDÍS
 	[^AOÁÉÓÚ] I N N		> 	-INN,FÍ
 	[^AOÁÉÓÚ] I N N		>	-INN,EADAR    # An B.N.
+	[^AOÁÉÓÚ] I N N		>	-INN,EAS      # An B.N.
 
 	[^HAOÁÉÓÚT] I N N	>	-INN,TEÁ      # gnáthchaite
 	[^HAOÁÉÓÚT] I N N	>	-INN,TÍ
@@ -529,6 +543,7 @@
 	É I N N		>	-NN,FIDÍS
 	É I N N		> 	-NN,FÍ
 	É I N N		>	-INN,ADAR             # An B.N.
+	É I N N		>	-INN,AS               # An B.N.
 
 					              # ghuínn, d'fhínn
 	Í N N		>	-NN,OMAR              # caite
@@ -542,6 +557,7 @@
 	Í N N		>	-NN,FIDÍS
 	Í N N		> 	-NN,FÍ
 	Í N N		>	-NN,ODAR              # An B.N.
+	Í N N		>	-NN,OS                # An B.N.
 
 	[^AOU] Í N N		>	-ÍNN,ITEÁ
 	[^AOU] Í N N		>	-ÍNN,ITÍ
@@ -609,6 +625,7 @@
 	A Í N N		>	-AÍNN,ÓIDÍS
 	A Í N N		>	-AÍNN,ÓFAÍ
 	A Í N N		>	-NN,ODAR        # An B.N. 
+	A Í N N		>	-NN,OS          # An B.N. 
 
                                                 # bhailínn, d'áitínn
 	[^A] Í N N	>	-NN,OMAR        # caite
@@ -624,6 +641,7 @@
 	[^A] Í N N	>	-ÍNN,EOIDÍS
 	[^A] Í N N	>	-ÍNN,EOFAÍ
 	[^A] Í N N	>	-NN,ODAR        # An B.N. 
+	[^A] Í N N	>	-NN,OS          # An B.N. 
 
 #  This syntax works, but unmunch doesn't like a 0 replacement:
 #   L A I M              >       -IM,""          # ainmfhocal briathartha
@@ -653,6 +671,7 @@
         L A I N N              >       -AINN,FAIDÍS
         L A I N N              >       -AINN,FAÍ
         L A I N N              >       -INN,DAR        # An B.N.
+        L A I N N              >       -INN,S          # An B.N.
                                                # *****>>>>>> 
 	L A I N N	       >       -LAINN,ILTEÁ    # gnáthchaite dara phearsa
 	L A I N N	       >       -LAINN,ILTÍ     # gnáthchaite saor
